util/test/Test/Pos/Util/TimerSpec.hs

```diff
@@ -0,0 +1,64 @@
+module Test.Pos.Util.TimerSpec
+    ( spec) where
+
+import           Universum
+import           Control.Concurrent.Async (async)
+import           Data.Time.Clock.POSIX (POSIXTime, getPOSIXTime)
+import           Data.Time.Units (Microsecond, fromMicroseconds, toMicroseconds)
+import           Test.Hspec (Spec, describe, it)
+import           Test.QuickCheck (Property, Gen, property, choose)
+import           Test.QuickCheck.Monadic (forAllM, monadicIO, run, assert)
+
+import           Pos.Util.Timer
+
+nominalDiffTimeToMicroseconds :: POSIXTime -> Microsecond
+nominalDiffTimeToMicroseconds = fromMicroseconds . round . (* 1000000)
+
+chooseMsc :: (Microsecond, Microsecond) -> Gen Microsecond
+chooseMsc (a, b) = fromMicroseconds <$> choose (toMicroseconds a, toMicroseconds b)
+
+-- | A timer should wait at least given interval.
+prop_wait :: Property
+prop_wait = 
+    let tMIN = 250 :: Microsecond
+        tMAX = 500 :: Microsecond
+    in monadicIO
+    $ forAllM (chooseMsc (tMIN, tMAX))
+    $ \t -> do
+        timer <- newTimer
+        startTime <- run getPOSIXTime
+        _ <- startTimer t timer
+        _ <- atomically $ waitTimer timer
+        endTime <- run getPOSIXTime
+        let diffTime = nominalDiffTimeToMicroseconds (endTime - startTime)
+        assert $ t <= diffTime 
+
+-- | A timer should be additive: (re)-starting a timer @t1@ after @t0@
+-- microseconds should wait @t0 + t1@ microseconds.
+prop_additive :: Property
+prop_additive = 
```

Comment:
This is a nice test but I think we should also include a similar one, in which a single timer is `startTimer`d twice: first with `t0` and then again with `t1` after `d`, _before_ `t0` has elapsed (say, `d = t0 / 2`, using a `threadDelay`). The test then checks that the observed elapsed time by the `waitTimer` thread is at least `t1 + d`

Reply:
I'll preserve the current implementation, because the test passes more consistently (with `threadDelay` I needed to run the timer for a longer period, which makes the test slower).